packages/grid/_modules_/grid/utils/utils.ts

```diff
@@ -15,7 +18,7 @@ export function formatDateToLocalInputDate({
   value: GridCellValue;
   withTime: boolean;
 }) {
-  if (isDate(value)) {
+  if (isDate(value) && isDateValid(value)) {
```

Comment:
```suggestion
  if (isDate(value)) {
```

Reply:
I would still leave it as if the same issue happen it's not creating an error

Reply:
If it's not creating an error, how will we know if the logic is wrong?

Reply:
We could enter this condition if we reuse this function in a custom component.
The question is, should the formatting throw an error for an invalid date or should it just not format anything?
The later seems better to me